### PR TITLE
Support ARM64 Visual Studio Cross Compiler

### PR DESCRIPTION
**Pull request by modmuss50**
Inital work towards completing: https://github.com/gradle/gradle/issues/21703
Required for: https://github.com/gradle/native-platform/pull/318

This is required to cross compile native-platform for ARM64 Windows from a x64/x86 host machine.

### Context
 This change will allow users to use the ARM64 MSVC cross compiler for their projects. This is also required to bootstrap native-platform on ARM64 windows.

Future work may include using the native ARM64 compiler on an ARM64 Windows host, as well as cross compiling for other contemporary platforms from an ARM64 host.

I look forward to your feedback. 👍 

Signed-off-by: modmuss50 <modmuss50@gmail.com>

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

**Comment by big-guy** [comment]
@bot-gradle test this

**Comment by VladRassokhin** [comment]
First I've missed your PR and almost submitted similar one :)

I'd also recommend to adjust `DefaultArchitecture#isArm` to `DefaultArchitecture#isArm32` with only `Architectures.ARM_V7.isAlias(name)` check there. That way you won't need to rely on `isArm64` check be done before `isArm` checks.

**Comment by modmuss50** [comment]
Hi, @VladRassokhin 

Thanks a lot for your interest in this and your suggestion. I have gone ahead and updated the PR 👍

**Comment by ThadHouse** [comment]
We're getting a lot of requests for Arm64 on Windows (We have mac and linux working for our setup). It'd be great if this could make 8.0, that way we can get closer to being able to fully run and build on arm64 windows.

**Comment by lptr** [comment]
Looks like Microsoft is ramping up ARM support on Windows, too: https://www.theverge.com/2022/10/24/23420207/microsoft-windows-dev-kit-2023-project-volterra-release-date-price

> The Windows Dev Kit 2023 hardware will help developers build Windows on Arm apps, just as Microsoft makes a bigger effort to improve its Arm efforts

File: subprojects/platform-native/src/main/java/org/gradle/nativeplatform/platform/internal/ArchitectureInternal.java [hunk 1]
```diff
@@ -31,5 +31,8 @@ enum InstructionSet { X86, ITANIUM, PPC, SPARC, ARM }
     boolean isIa64();
 
     @Internal
-    boolean isArm();
```
Comment (big-guy) [inline subprojects/platform-native/src/main/java/org/gradle/nativeplatform/platform/internal/ArchitectureInternal.java:34]: Could you keep this (for backwards compatibility)?  We'd have three methods then:

isArm - true when on ARM (32 or 64)
isArm32 - true when on ARM 32-bit only
isArm64 - true when on ARM 64-bit only

Comment (modmuss50) [reply subprojects/platform-native/src/main/java/org/gradle/nativeplatform/platform/internal/ArchitectureInternal.java:34]: Done 👍

Comment (VladRassokhin) [reply subprojects/platform-native/src/main/java/org/gradle/nativeplatform/platform/internal/ArchitectureInternal.java:34]: Sorry for a bit stupid question, but what's the purpose of preserving backward compatibility in class with 'Internal' in name in a package with 'internal' in name?
Is there any data that it's used outside of this repo?

Comment (ThadHouse) [reply subprojects/platform-native/src/main/java/org/gradle/nativeplatform/platform/internal/ArchitectureInternal.java:34]: Groovy can access these properties, and I think at one point some parts of the c++ documentation used them. So it’s better to not explicitly break anyone using it, especially when keeping it working is fairly easy.

**Comment by ThadHouse** [comment]
Any chance this could make it into 7.6, and not all the way into 8.0? Since its going to require a change to native-platform to be able to provide arm64 native binaries, I feel like it'd be nice to get those arm64 binaries into 8.0, which would require these changes in a prior release, or for native-platform to build with a non release version of gradle.

**Comment by ThadHouse** [comment]
This is such a simple change? Why does it keep getting pushed back?

**Comment by ThadHouse** [comment]
Ping @big-guy @lptr @marcphilipp 
This is a really simple review, and every time it gets pushed back or ignored it just enforces the point that gradle doesn't care about Native anymore.

**Pull request merged**